lib/modules/datasource/rubygems/get.ts

```diff
@@ -182,4 +200,87 @@ export class InternalRubyGemsDatasource extends Datasource {
 
     return new Marshal(response.body).parsed as T;
   }
+
+  private async isNexusDataSource(registry: string): Promise<boolean> {
+    const statusEndPoint = '/service/rest/v1/status';
+    const endPointRe = regEx(`^(?<endPoint>.*:\\d{1,5})\\/`);
+    const { endPoint } = registry.match(endPointRe)?.groups ?? {
+      endPoint: '',
+    };
+    const nexusUrl = endPoint.concat(statusEndPoint);
+    let isNexus = false;
+    try {
+      const response = await this.http.getJson<HttpResponse>(nexusUrl);
+      isNexus = response.headers?.server?.includes('Nexus') ?? false;
+    } catch (err) {
+      logger.debug({ err }, 'Not a nexus datasource');
+    }
+    return isNexus;
+  }
```

Comment:
in case you failed the regex, why do we continue to call http URL?
isn't that an indication of failure?
shouldn't we stop there and skip the external call?